### PR TITLE
Feature/ether erc20

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Add ethereum ERC20 token deposit support

### CODE REVIEW

File: src/main/kotlin/notary/NotaryImpl.kt [hunk 1]
```diff
@@ -21,51 +22,60 @@ class NotaryImpl(
...
+    private fun onEthSidechainDeposit(
+        hash: String,
+        account: String,
+        asset: String,
+        amount: BigInteger
+    ): IrohaOrderedBatch {
+        val domain = "ethereum"
 
-        val destAccountId = ethInputEvent.input
+        logger.info { "transfer Ethereum event:" }
```
Comment (x3medima17) [inline src/main/kotlin/notary/NotaryImpl.kt]: It is better to use one logger invocation in this case

File: src/main/kotlin/notary/NotaryImpl.kt [hunk 2]
```diff
@@ -79,10 +89,18 @@ class NotaryImpl(
...
-
-        // TODO replace output with effective implementation
-            else -> IrohaOrderedBatch(arrayListOf())
+            is NotaryInputEvent.EthChainInputEvent.OnEthSidechainDeposit -> onEthSidechainDeposit(
+                ethInputEvent.hash,
+                ethInputEvent.user,
+                ethereumAssetId,
+                ethInputEvent.amount
+            )
+            is NotaryInputEvent.EthChainInputEvent.OnEthSidechainDepositToken -> onEthSidechainDeposit(
```
Comment (x3medima17) [inline src/main/kotlin/notary/NotaryImpl.kt]: Code duplication

File: src/main/kotlin/sideChain/eth/EthChainHandler.kt [hunk 3]
```diff
@@ -1,19 +1,86 @@
...
+                        wallets.containsKey(to)
+            }
+            .map {
+                // second and third topics are addresses from and to
+                val from = "0x" + it.topics[1].drop(26).toLowerCase()
+                val to = "0x" + it.topics[2].drop(26).toLowerCase()
+                // amount of transfer is stored in data
+                val amount = BigInteger(it.data.drop(2), 16)
+
+                val userId = wallets[to]
```
Comment (x3medima17) [inline src/main/kotlin/sideChain/eth/EthChainHandler.kt]: If value of a map may be null, I would use `Nullable` type

File: src/main/kotlin/sideChain/eth/EthChainHandler.kt [hunk 4]
```diff
@@ -1,19 +1,86 @@
...
+        val userId = wallets[tx.to]
+        if (userId == null) {
+            logger.error { "transfer to unknown address ${tx.to}" }
+            return listOf()
+        }
+
+        return listOf(
+            NotaryInputEvent.EthChainInputEvent.OnEthSidechainDeposit(
+                tx.hash,
+                wallets[tx.to]!!,
```
Comment (x3medima17) [inline src/main/kotlin/sideChain/eth/EthChainHandler.kt]: Are you sure about this code?

File: src/test/notary/NotaryTest.kt [hunk 5]
```diff
@@ -57,33 +47,42 @@ class NotaryTest {
...
+                        cmd = createAsset.first()
+                        if (cmd is IrohaCommand.CommandCreateAsset) {
+                            assertEquals(expectedAssetId, cmd.assetName)
+                            assertEquals("ethereum", cmd.domainId)
+                            assertEquals(0, cmd.precision)
                         } else {
                             fail { "Wrong IrohaCommand type" }
                         }
 
                         val transferAsset = txs[2].commands
```
Comment (x3medima17) [inline src/test/notary/NotaryTest.kt]: Please fix naming

File: src/main/kotlin/notary/NotaryImpl.kt (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply src/main/kotlin/notary/NotaryImpl.kt]: Different input events call function with different parameters (ethereumAssetId and ethInputEvent.token).

**Pull request merged**